url433bec1e56_sarobe_DifficultyInvestigation_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/fr/inria/optimization/cmaes/CMAEvolutionStrategy.java

```diff
@@ -0,0 +1,2704 @@
+         *    Example: max(Double arx, arx[0]) */
+        public Double max(Double ar[], Comparator<Double> c) {
+            int i;
+            Double m;
+            m = ar[0];
+            for (i = 1; i < ar.length; ++i) {
+                if (c.compare(m, ar[i]) > 0)
+                    m = ar[i];
+            }
+            return m;
+        }
+        
+        /** @return Maximum value of 1-D IntDouble array */
+        public IntDouble max(IntDouble ar[]) {
+            int i;
+            IntDouble m;
+            m = ar[0];
+            for (i = 1; i < ar.length; ++i) {
+                if (m.compare(m, ar[i]) < 0)
+                    m = ar[i];
+            }
+            return m;
+        }
+        
+        /** @return Minimum value of 1-D IntDouble array */
+        public IntDouble min(IntDouble ar[]) {
+            int i;
+            IntDouble m;
+            m = ar[0];
+            for (i = 1; i < ar.length; ++i) {
+                if (m.compare(m, ar[i]) > 0)
+                    m = ar[i];
+            }
+            return m;
+        }
+        
+        /** @return Minimum value of 1-D Object array defining a Comparator */
+        public Double min(Double ar[], Comparator<Double> c) {
+            int i;
+            Double m;
+            m = ar[0];
+            for (i = 1; i < ar.length; ++i) {
+                if (c.compare(m, ar[i]) < 0)
+                    m = ar[i];
+            }
+            return m;
+        }
+        
+        /**
+         * @return Diagonal of an 2-D double array
+         */
+        public double[] diag(double ar[][]) {
+            int i;
+            double[] diag = new double[ar.length];
+            for (i = 0; i < ar.length && i < ar[i].length; ++i)
+                diag[i] = ar[i][i];
+            return diag;
+        }
+        
+        /**
+         * @return 1-D double array of absolute values of an 1-D double array
+         */
+        public double[] abs(double v[]) {
+            double res[] = new double[v.length];
+            for(int i = 0; i < v.length; ++i)
+                res[i] = Math.abs(v[i]);
+            return res;
+        }
+    } // MyMath
+    
+    class Timing {
+        Timing(){
+            birth = System.currentTimeMillis();
+            start = birth; // on the save side 
+        }
+        long birth; // time at construction, not really in use
+        long start; // time at end of init()
+        long starteigen; // time after flgdiag was turned off, ie when calls to eigen() start
+        long eigendecomposition = 0; // spent time in eigendecomposition
+        long writedefaultfiles = 0;        // spent time in writeToDefaultFiles
+    }
+    Timing timings = new Timing();
+    
+    /* flgforce == 1 force independent of time measurments, 
+     * flgforce == 2 force independent of uptodate-status
+     */
+    void eigendecomposition(int flgforce) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.